tests/labels_test.go

```diff
@@ -0,0 +1,130 @@
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"kubevirt.io/ssp-operator/internal/common"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"gomodules.xyz/jsonpatch/v2"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func expectedLabelsFor(name string, component common.AppComponent) map[string]string {
+	return map[string]string{
+		common.AppKubernetesNameLabel:      name,
+		common.AppKubernetesManagedByLabel: "ssp-operator",
+		common.AppKubernetesPartOfLabel:    strategy.GetPartOfLabel(),
+		common.AppKubernetesVersionLabel:   strategy.GetVersionLabel(),
+		common.AppKubernetesComponentLabel: component.String(),
+	}
+}
+
+func expectAppLabels(res *testResource) {
+	testExpectedLabels(res.NewResource(), res.GetKey(), res.ExpectedLabels)
+}
+
+func expectAppLabelsRestoreAfterUpdate(res *testResource) {
+	testExpectedLabelsRestoreAfterUpdate(res.NewResource(), res.GetKey(), res.ExpectedLabels)
+}
+
+func testExpectedLabels(resource controllerutil.Object, key client.ObjectKey, expectedLabels map[string]string) {
+	patchAppLabelsIntoSSP(expectedLabels)
+	waitForLabelMatch(resource, key, expectedLabels)
+}
+
+func testExpectedLabelsRestoreAfterUpdate(resource controllerutil.Object, key client.ObjectKey, expectedLabels map[string]string) {
+	testExpectedLabels(resource, key, expectedLabels)
+
+	operations := newLabelOperations(expectedLabels)
+	for label := range expectedLabels {
+		operations = append(operations, labelPatchOperationFor("replace", label, "wrong"))
+	}
+	patch := encodePatch(operations)
+	err := apiClient.Patch(ctx, resource, patch)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(labelsMatch(expectedLabels, resource)).To(BeFalse())
+}
```

Comment:
This function `testExpectedLabelsRestoreAfterUpdate` is not actually testing that the labels are restored after update. Please add an `Eventually()` block that will check it.